ote_sdk/ote_sdk/utils/argument_checks.py

```diff
@@ -0,0 +1,217 @@
+"""
+Utils for checking functions and methods arguments
+"""
+
+# Copyright (C) 2021-2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+import typing
+from os.path import exists
+
+from numpy import floating
+from omegaconf import DictConfig
+from yaml import safe_load
+
+
+def raise_value_error_if_parameter_has_unexpected_type(
+    parameter, parameter_name, expected_type
+):
+    """Function raises ValueError exception if parameter has unexpected type"""
+    if not isinstance(parameter, expected_type):
+        parameter_type = type(parameter)
+        raise ValueError(
+            f"Unexpected type of '{parameter_name}' parameter, expected: {expected_type}, actual: {parameter_type}"
+        )
+
+
+def check_parameter_type(parameter, parameter_name, expected_type):
+    """Function extracts nested expected types and raises ValueError exception if parameter has unexpected type"""
+    # pylint: disable=W0212
+    if isinstance(expected_type, typing._GenericAlias):  # type: ignore
```

Comment:
1. First of all, we have there a very big `if`, and one of branch of the `if` is much shorter than the other.
   In this case I really like use approach "quick handle simple branch and return", it allows to decrease indent level for most of the function, it forms the code as follows:
   ```python
   if not isinstance(expected_type, typing._GenericAlias):
         quick_handle_short_branch(...)
         return
   ```

2. As we spoke, we need special checks for paths (no null chars, no empty str, no non-printable characters). Formally, they are just `str`, but we could make special type of checks just here. We could make a special base class and special derived class for paths:
   ```python
   from abc import ABC, abstractmethod
   class BaseInputArgumentChecker(ABC):
       @abstractmethod
       def check(self, var):
           raise NotImplementedError('The check is not implemented')
   class PathChecker(BaseArgumentChecker):
       def check(self, var, var_name):
           raise_value_error_if_parameter_has_unexpected_type(var, ...., str)
           check_path_no_null_char(var, var_name)
           check_path_no_nonprintable_chars(var, var_name)
   ```
   After creating these classes in the top of this file you will be able to add branch in this function
   ```python
       if isinstance(expected_type, BaseInputArgumentChecker):
           expected_type.check(parameter, parameter_name)
           return
   ```
   After that you will have to use PathChecker everywhere where it is required. 🙂